migrations/2019_10_22_000000_change_reason_text_col_type.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+/*
+ * This file is part of Flarum.
+ *
+ * (c) Toby Zerner <toby.zerner@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Database\Schema\Builder;
+
+return [
+    'up' => function (Builder $schema) {
+        $schema->table('flags', function (Blueprint $table) {
+            $table->text('reason_detail')->change();
+        });
+    },
+
+    'down' => function (Builder $schema) {
+        $schema->table('flags', function (Blueprint $table) {
+            $table->string(['reason_detail'])->change();
+        });
+    }
+];
```

Comment:
We decided we would prefer to keep the current `string` field. That should be long enough for a simple explanation; and we can enforce it with a `maxlength` attribute and a validation.

Reply:
I'd put forward that a field `varchar` with a max of 255 is not enough, as we have found.

Use case: a user flagging a post as a duplicate of another, and provides a url along with their comments. 

Might I suggest a combination of your suggestion and mine? Let's keep the migration to `text`, but add a `maxlength` variable, exposed to the extension setting? This will allow admin to select their own preference, based on their community needs.

Reply:
Hmm, I guess that's convincing. cc @luceos @datitisev 

No need for the config for now, I'd say.